### PR TITLE
[WholeGraph] Refactor code to separate WholeGraph-related functions

### PR DESCRIPTION
**Pull request by chang-l**
This pull request refactors the code to separate WholeGraph-related functions into a separate module. This improves code organization and makes it easier to work with/extend WholeGraph functionality.

### CODE REVIEW

File: python/graphstorm/dataloading/dataset.py [hunk 1]
```diff
@@ -26,9 +26,11 @@
...
 import pandas as pd
 
-from ..utils import get_rank, get_world_size, is_distributed, barrier
-from ..utils import sys_tracker, is_wholegraph
-from .utils import dist_sum, flip_node_mask, is_wholegraph_embedding
+from ..utils import get_rank, get_world_size, is_distributed, barrier, is_wholegraph
+from ..utils import sys_tracker
+from .utils import dist_sum, flip_node_mask
+
+from ..wholegraph_utils import is_wholegraph_embedding
```
Comment (classicsong) [inline python/graphstorm/dataloading/dataset.py:33]: Can we have a whole graph sub-module under graphstorm?
Then we can do `from ..wholegraph import xxx` (or say `from graphstorm.wholegraph import xxx`) and `from ..wholegraph .utils import xxx`.

Comment (chang-l) [reply python/graphstorm/dataloading/dataset.py]: Moved wholegraph-related functions/classes to a sub-package `graphstorm.wholegraph`. Is this what you expected?

File: python/graphstorm/wholegraph/utils.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+"""This module provides utility functions for working with WholeGraph"""
```
Comment (classicsong) [inline python/graphstorm/wholegraph/utils.py:1]: Please add a Copyright

File: python/graphstorm/wholegraph/wholegraph.py [hunk 3]
```diff
@@ -0,0 +1,268 @@
+"""This module provides functionality for working with the WholeGraph"""
```
Comment (classicsong) [inline python/graphstorm/wholegraph/wholegraph.py:1]: ditto

**Review by classicsong — APPROVED** [review]

**Pull request merged**